.github/actions/on_host_test/action.yaml

```diff
@@ -59,6 +59,8 @@ runs:
       shell: bash
       run: |
         echo "PYTHONPATH=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+        echo "TEST_RESULTS_DIR=${GITHUB_WORKSPACE}/unit-test-results" >> $GITHUB_ENV
```

Comment:
Consider using step inputs/outputs instead of environment variables for this

Reply:
FYI https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Reply:
imo it's better for the better validation (the env doesn't care if something actually gets set while the output does) and more explicit notation. Igor knows more about it though, so if he doesn't care about which way we do it then I don't

Reply:
I think rule of thumb is to use env file to share data between steps and env output to share data between jobs.

Reply:
Another consideration is that outputs don't work correctly for matrix jobs. The matrix jobs all write to the same output so only the last one is saved.

It's not relevant for this CL since the value is not dependent on the matrix values.